orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java

```diff
@@ -0,0 +1,559 @@
+			if ( !entityQueries.containsKey( entityKeyMetadata ) ) {
+				// Embeddables metadata
+				entityQueries.put( entityKeyMetadata, new OrientDBEntityQueries( entityKeyMetadata ) );
+			}
+		}
+		return entityQueries;
+	}
+
+	private Map<EntityKeyMetadata, OrientDBEntityQueries> initializeEntityQueries(SessionFactoryImplementor sessionFactoryImplementor) {
+		Map<EntityKeyMetadata, OrientDBEntityQueries> queryMap = new HashMap<EntityKeyMetadata, OrientDBEntityQueries>();
+		Collection<EntityPersister> entityPersisters = sessionFactoryImplementor.getEntityPersisters().values();
+		for ( EntityPersister entityPersister : entityPersisters ) {
+			if ( entityPersister instanceof OgmEntityPersister ) {
+				OgmEntityPersister ogmEntityPersister = (OgmEntityPersister) entityPersister;
+				queryMap.put( ogmEntityPersister.getEntityKeyMetadata(), new OrientDBEntityQueries( ogmEntityPersister.getEntityKeyMetadata() ) );
+			}
+		}
+		return queryMap;
+	}
+
+	private Map<AssociationKeyMetadata, OrientDBAssociationQueries> initializeAssociationQueries(SessionFactoryImplementor sessionFactoryImplementor) {
+		Map<AssociationKeyMetadata, OrientDBAssociationQueries> queryMap = new HashMap<AssociationKeyMetadata, OrientDBAssociationQueries>();
+		Collection<CollectionPersister> collectionPersisters = sessionFactoryImplementor.getCollectionPersisters().values();
+		for ( CollectionPersister collectionPersister : collectionPersisters ) {
+			if ( collectionPersister instanceof OgmCollectionPersister ) {
+				OgmCollectionPersister ogmCollectionPersister = (OgmCollectionPersister) collectionPersister;
+				log.info( "initializeAssociationQueries: ogmCollectionPersister :" + ogmCollectionPersister );
+				EntityKeyMetadata ownerEntityKeyMetadata = ( (OgmEntityPersister) ( ogmCollectionPersister.getOwnerEntityPersister() ) ).getEntityKeyMetadata();
+				log.info( "initializeAssociationQueries: ownerEntityKeyMetadata :" + ownerEntityKeyMetadata );
+				AssociationKeyMetadata associationKeyMetadata = ogmCollectionPersister.getAssociationKeyMetadata();
+				log.info( "initializeAssociationQueries: associationKeyMetadata :" + associationKeyMetadata );
+				queryMap.put( associationKeyMetadata, new OrientDBAssociationQueries( ownerEntityKeyMetadata, associationKeyMetadata ) );
+			}
+		}
+		return queryMap;
+	}
+
+	@Override
+	public GridType overrideType(Type type) {
+		log.info( "overrideType:" + type.getName() + ";" + type.getReturnedClass() );
+		GridType gridType = null;
+		if ( type.getName().equals( "com.orientechnologies.orient.core.id.ORecordId" ) ) {
+			gridType = ORecordIdGridType.INSTANCE;
+		} else if ( type.getName().equals( "com.orientechnologies.orient.core.db.record.ridbag.ORidBag" ) ) {
+			gridType = ORidBagGridType.INSTANCE;
+		}
+		else {
+			gridType = super.overrideType( type ); // To change body of generated methods, choose Tools | Templates.
```

Comment:
If it's possible, it would be nice if you could send PRs without this kind of comments.